modules/barcode-scan/src/main/java/com/gluonhq/attach/barcodescan/BarcodeScanService.java

```diff
@@ -99,11 +100,8 @@ static Optional<BarcodeScanService> create() {
      * Starts up the scanner functionality (commonly provided via the camera), and then parsed by Attach to
      * determine the string the barcode represents.
      *
-     * @return Returns an Optional containing the parsed string. The Optional may
-     *         be empty if the String fails to be parsed for any reason, or if the
-     *         user cancels the operation.
      */
-    Optional<String> scan();
+    void scan();
```

Comment:
This would be a breaking change. It's probably better to deprecate this method and add new API. 

Possible names are `nonBlockingScan()`, `startScan()`, `asyncScan()`... or any other one you propose.

(If we choose `startScan()` a follow up issue could be adding a `stopScan()` method that could be called to programmatically stop the camera view, if that makes sense).

Also add `@since`, like:
```
@since 4.0.16
void xxxScan();
```

Reply:
I like `startScan`. But do you really think there is need for a `stopScan` method? I mean, I think once the scan view is open it is up to the user to cancel it (if no code can be detected etc). For that on iOS there is the native `cancel` method and on Android I think the user just uses the "Back" soft-key, no?

Reply:
Yes, that is the problem with this naming, `start` implies that there is a `stop`, and it it not really (unless we want to programmatically stop the camera view, which shouldn't really happen).

It is probably better then to use `asyncScan()`.

Reply:
Yes, I think you're right. So we'll go with `asyncStart` then.

Reply:
What do you suggest we do for the deprecated implementation methods? For those ones we still need the `exitNestedLoop` call in `public static void setResult(String v)`. Is it ok to calll this if there never was a `enterNestedLoop` when our new methods are used?

Reply:
maybe just add a static boolean `enteredLoop` or so, set it to true if the old scan method calls `enterNestedLoop` and then check if it is true in `setResult`.

Reply:
I just pushed the changes. Can you go from here or is there something I have to do first?